src/xo-app/backup/restore/index.js

```diff
@@ -98,16 +103,88 @@ const doImport = ({ backup, sr, start }) => {
   }
 }
 
+class _Collapsible extends Component {
+  render () {
+    const {
+      collapsible,
+      children,
+      ...props
+    } = this.props
+
+    return collapsible
+      ? <Collapse {...props}>{children}</Collapse>
+      : <div>
+        <span>{props.buttonText}</span>
+        <br />
+        {children}
+      </div>
+  }
+}
+
 class _ModalBody extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      mapVdisSrs: {}
+    }
+  }
+
   get value () {
     return this.state
   }
 
+  _getSrPredicate = createSelector(
+    () => this.state.sr,
+    () => this.state.mapVdisSrs,
+    (defaultSr, mapVdisSrs) => sr => {
+      let result = isSrWritable(sr) && defaultSr.$pool === sr.$pool
+
+      if (!defaultSr.shared) {
+        result = result && (sr.shared || defaultSr.$container === sr.$container)
+      } else {
+        forEach(mapVdisSrs, selectedSr => {
+          if (selectedSr != null && !selectedSr.shared) {
+            result = result && (sr.shared || selectedSr.$container === sr.$container)
+          }
+        })
+      }
+
+      return result
+    }
+  )
+
+  _onChangeDefaultSr = event => {
+    const oldSr = this.state.sr
+    const newSr = getEventValue(event)
+
+    if (oldSr == null || newSr == null || oldSr.$pool !== newSr.$pool) {
+      this.setState({
+        mapVdisSrs: {}
+      })
+    } else if (!newSr.shared) {
+      const mapVdisSrs = {...this.state.mapVdisSrs}
+      forEach(mapVdisSrs, (sr, vdi) => {
+        if (sr != null && newSr !== sr && !sr.shared) {
```

Comment:
If `sr` isn't shared but is on the same host as `newSr`, it shouldn't be removed, right?